client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -92,6 +99,22 @@ export const useStepNavigationWithTracking = ( {
 					stepNavigation.exitFlow?.( to );
 				},
 			} ),
+			/**
+			 * If the `previousStep` is defined in the store, it's a solid proxy to guess that we navigated at least once via Stepper's React Router.
+			 * If the flow doesn't define a `goBack` handler, and `previousStep` is defined, we can just go history.back() and we'll remain in the flow.
+			 * But if `previousStep` is not defined, and the flow doesn't define a `goBack` handler, we should return undefined so the StepContainer doesn't render a back button.
+			 */
+			...( previousStep && {
+				goBack: () => {
+					handleRecordStepNavigation( {
+						event: STEPPER_TRACKS_EVENT_STEP_NAV_GO_BACK,
+					} );
+					history.back();
```

Comment:
I suppose another option could be to actually go to the previous step explicitly like `navigate( previousStep )` - not real code! I'm not sure which is better...

Reply:
This was my initial approach, but it's bad because it pushes a new state down the browser's history stack when we want to pop. When I called `navigate`, going back via the browser went forward 🌀.